### PR TITLE
Remove addon performance testing, unused for years (issue #1058)

### LINKED ISSUE #1058: Remove perf
Performance testing of add-ons is no more. This is a telling comment:

```
# These are all the apps available for a file to be tested against:
ALL_APPS = ('firefox3.6', 'firefox6.0')
```

Lets remove `apps/devhub/perf.py` and everything related to it.

### PR DESCRIPTION
**Pull request by magopian**
Fixes #1058

This removes a _lot_ of stuff, in a _lot_ of files, so it would make sense to have QA use this for as long as possible on -dev and stage.

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
It appears you forgot to update asset-related configs, besides that - although it's hard to review a PR with only deletions it's at least a :+1: with :beers: and :cake: :laughing:

File: apps/perf/models.py [hunk 1]
```diff
@@ -1,94 +0,0 @@
...
-            log.info('startup_is_too_slow() only applies to startup time, '
-                     'not %s' % self.test)
-            return False
-        if not baseline:
-            baseline = self.get_baseline()
-        delta = (self.average - baseline) / baseline * 100
-        return delta >= self.get_threshold()
-
-    class Meta:
-        db_table = 'perf_results'
```
Comment (andymckay) [inline apps/perf/models.py]: The tables mentioned in this can be removed if they exist?

Comment (magopian) [inline apps/perf/models.py]: ah yes, very good point! added to the migration

File: apps/api/templates/api/includes/addon.xml [hunk 2]
```diff
@@ -134,20 +134,9 @@
     <homepage>{{ addon.homepage }}</homepage>
     <support>{{ addon.support_url }}</support>
     <featured>{{ addon.is_featured(request.APP, request.LANG)|int }}</featured>
-    {% if settings.API_SHOW_PERF_DATA %}
+    {# Only kept for backwards compatibility of the API #}
```
Comment (EnTeQuAk) [inline apps/api/templates/api/includes/addon.xml]: Do we want to keep that around forever?

**Comment by EnTeQuAk** [comment]
Can we drop the waffle-flag from the database too?

**Comment by EnTeQuAk** [comment]
r+wc

File: apps/api/templates/api/includes/addon.xml (hunk 2 above)
Comment (magopian) [inline apps/api/templates/api/includes/addon.xml]: Well, at least until we deprecate the current API which uses that.

**Comment by magopian** [comment]
Very good point for the waffle flag, I've removed it, thanks!

**Pull request merged**